csrc/id_model/contiguity.cpp

```diff
@@ -86,6 +86,16 @@ ContigIDGroups::ContigIDGroups(
 void ContigIDGroups::handle(Merge* merge, Direction direction) {
   // Only forward direction is supported for now
   if (direction != Direction::Forward) {
+    // Backward propagation for Merge is not implemented.
```

Comment:
Why the backward propagation of merge is skipped, but not forward of split? Is it because in practice, we could often see things like

```C++
// I0    I1  allocation
//  \    / \
//   \ I2   I3
//    \/   /
//    I4  /
//      \/
//      I5  loop
```
but not the following?

```C++
//      I5 loop
//      /\
//    I4  \
//    /\   \
//   / I2  I3
//  /    \ /
// I0     I1 allocation
```

Reply:
It's just because traversing a Merge backward won't help simplify indices. In the second example case above, if we analyzed a backward Merge like `I2, I3 -> I1`, the indexing traversal would stop at `I2` and `I3` instead of `I1`. The end result should be the same, however, since both of them would result in something like `idx2 * i3_extent + idx3`.

The forward split should also be the same. I don't think it would contribute to index simplification. I probably just didn't realize it.